src/circular-progress.jsx

```diff
@@ -1,10 +1,75 @@
 import React from 'react';
-import ReactDOM from 'react-dom';
-import StylePropable from './mixins/style-propable';
 import autoPrefix from './styles/auto-prefix';
 import Transitions from './styles/transitions';
 import getMuiTheme from './styles/getMuiTheme';
 
+function getRelativeValue(value, min, max) {
+  const clampedValue = Math.min(Math.max(min, value), max);
+  const rangeValue = max - min;
+  const relValue = Math.round(clampedValue / rangeValue * 10000) / 10000;
+  return relValue * 100;
+}
+
+function getStyles(props, state) {
+  const {
+    max,
+    min,
+    size,
+    value,
+  } = props;
+
+  const {
+    baseTheme: {
+      palette,
+    },
+  } = state.muiTheme;
+
+  const zoom = size * 1.4 ;
+  const baseSize = '50px';
+  let margin = Math.round( ((50 * zoom) - 50) / 2 );
+
+  if (margin < 0) margin = 0;
+
+  let styles = {
+    root: {
+      position: 'relative',
+      margin: margin,
+      display: 'inline-block',
+      width: baseSize,
+      height: baseSize,
+    },
+    wrapper: {
+      width: baseSize,
+      height: baseSize,
+      display: 'inline-block',
+      transition: Transitions.create('transform', '20s', null, 'linear'),
+    },
+    svg: {
+      height: baseSize,
+      position: 'relative',
+      transform: `scale(${zoom})`,
+      width: baseSize,
+    },
+    path: {
+      strokeDasharray: '89,200',
+      strokeDashoffset: 0,
+      stroke: props.color || palette.primary1Color,
+      strokeLinecap: 'round',
+      transition: Transitions.create('all', '1.5s', null, 'ease-in-out'),
+    },
+  };
+
+  autoPrefix.set(styles.wrapper, 'transitionTimingFunction', 'linear', state.muiTheme);
```

Comment:
Yeah good call. :+1:  After the migration I'd like to revisit seeing if we can get rid of all the `autoPrefix` calls...